client-api: add unstable MSC3266 support (room summary) (#1776)
ruma/ruma: crates/ruma-client-api/CHANGELOG.md

- Add `dir` `Request` field on the `get_relating_events_with_rel_types` and
  `get_relating_events_with_rel_type_and_event_type` endpoints
- Add unstable support for moderator server support discovery, according to MSC4121

> - Add unstable support for MSC3266 behind the `unstable-msc3266` feature, and a new
>   `room::get_summary` module.

Reviewer: This changelog entry makes it sound like the unstable MSC3266 and `room::get_summary` are separate.
```suggestion
- Add unstable support for the room summary endpoint from MSC3266 behind the
  `unstable-msc3266` feature.
```
